projects/dotcms-field-elements/src/components/dot-binary-file/dot-binary-file.tsx

```diff
@@ -0,0 +1,312 @@
+
+    /** (optional) Disables field's interaction */
+    @Prop({ reflectToAttr: true })
+    disabled = false;
+
+    /** (optional) Describes a type of file that may be selected by the user, separated by comma  eg: .pdf,.jpg  */
+    @Prop({ reflectToAttr: true })
+    accept = '';
+
+    /** (optional) Text that be shown in the browse file button */
+    @Prop({ reflectToAttr: true })
+    buttonLabel = 'Browse';
+
+    @State() status: DotFieldStatus;
+    @State() errorElement: JSX.Element = null;
+
+    @Event() valueChange: EventEmitter<DotFieldValueEvent>;
+    @Event() statusChange: EventEmitter<DotFieldStatusEvent>;
+
+    allowedFileTypes = [];
+
+    /**
+     * Reset properties of the field, clear value and emit events.
+     */
+    @Method()
+    reset(): void {
+        this.value = '';
+        this.status = getOriginalStatus(this.isValid());
+        this.emitStatusChange();
+        this.emitValueChange();
+    }
+
+    componentWillLoad(): void {
+        this.validateProps();
+        this.status = getOriginalStatus(this.isValid());
+        this.emitStatusChange();
+    }
+
+    componentDidLoad(): void {
+        this.dragAndDropHandler();
+        this.pasteHandler();
+    }
+
+    @Watch('accept')
+    optionsWatch(): void {
+        const validTypes = checkProp<DotBinaryFileComponent, string>(this, 'accept');
+        this.allowedFileTypes = !!validTypes ? validTypes.split(',') : [];
+    }
+
+    hostData() {
+        return {
+            class: getClassNames(this.status, this.isValid(), this.required)
+        };
+    }
+
+    render() {
+        return (
+            <Fragment>
+                <dot-label
+                    label={this.label}
+                    required={this.required}
+                    name={this.name}
+                    tabindex="0"
+                >
+                    <input
+                        type="text"
+                        onBlur={() => this.blurHandler()}
+                        disabled={this.disabled}
+                        placeholder={this.placeholder}
+                        onKeyDown={(event: KeyboardEvent) => this.keyDownHandler(event)}
+                        onKeyPress={(event: KeyboardEvent) => event.preventDefault()}
+                    />
+                    <input
+                        aria-describedby={getHintId(this.hint)}
+                        class={getErrorClass(this.status.dotValid)}
+                        disabled={this.disabled}
+                        id={getId(this.name)}
+                        onChange={(event: Event) => this.changeHandler(event)}
+                        required={this.required || null}
+                        type="file"
+                        accept={this.accept}
+                        value={this.value}
+                    />
+                    <button disabled={this.disabled} onClick={() => this.buttonHandler()}>
+                        {this.buttonLabel}
+                    </button>
```

Comment:
why do we need this button here?

Reply:
this button is in charge of file the `input[type=file]` click event, to avoid show the default input, and display the file name in the `dot-binary-text-field`